### PR TITLE
Fix/buf optimisation

### PR DESCRIPTION
**Pull request by kepkap**
the issue: https://github.com/openmarketplaceengine/openmarketplaceengine/issues/137

1. makefile cleanup/fixes
2. gitignoring
3. linter issues

### CODE REVIEW

**Review by vlkolmakov — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -51,24 +51,15 @@ PACKAGES=$(shell go list ./...)
 PACKAGES_WITH_TESTS = $(shell go list -f '{{if len .XTestGoFiles}}{{.ImportPath}}{{end}}' ./... \
 							&& go list -f '{{if len .TestGoFiles}}{{.ImportPath}}{{end}}' ./...)
 
-checkstyle: ## Run quick checkstyle (govet + goimports (fail on errors))
-	@echo "==> Running govet"
-	go vet $(PACKAGES) || exit 1
-	@echo "==> Running goimports"
-	goimports -l -w .
-	@echo "==> SUCCESS";
```
Comment (kevinmichaelchen) [inline Makefile:59]: Why remove this?

File: .gitignore [hunk 2]
```diff
@@ -2,4 +2,7 @@
 .idea
 build
 cover.*
-node_modules
\ No newline at end of file
+node_modules
+
+validate
+internal/validate
```
Comment (kevinmichaelchen) [inline .gitignore:8]: I think we don't need this line.

See this [snippet](https://github.com/kevinmichaelchen/api-go-template/blob/bcd3384493b31a8606751350cc9acf41d153e0bf/Makefile#L6-L13).

We remove `validate` before we generate. We only bring it back (with `buf export` once the generation is complete).

File: Makefile [hunk 3]
```diff
@@ -104,24 +94,17 @@ lint_install: ## Install linter
...
 lint: ## Run linter
-	@echo "==> Running linter"
 	golangci-lint run
 
-gen-proto:
-	buf breaking --against '.git#branch=main'
-	buf lint
+buf-gen: ## Run buf lint format generate
+	buf export buf.build/envoyproxy/protoc-gen-validate -o .
+	# Not tracking breaking changes just yet, uncomment after first major release.
```
Comment (kevinmichaelchen) [inline Makefile:101]: 👍🏼

File: Makefile (hunk 1 above)
Comment (kepkap) [reply Makefile:59]: goimports is included in linter we use. so this target is just a dup in functionality.

File: .gitignore (hunk 2 above)
Comment (kepkap) [reply .gitignore:8]: now i got it. yep, will use the example from the snippet

Comment (kepkap) [reply .gitignore:8]: here is the resulting buf-gen
```
rm -rf api/{google,grafeas,validate} || true
	buf mod update api
	buf export buf.build/envoyproxy/protoc-gen-validate -o api

	# Proto files and types from google and grafeas
	# buf export buf.build/googleapis/googleapis -o api

	# Not tracking breaking changes just yet, uncomment after first major release.
	# buf breaking --against '.git#branch=main'
	buf lint --exclude-path api/validate/validate.proto
	buf format -w
	buf generate
```

**Pull request merged**